blaze/compute/numpy.py

```diff
@@ -197,9 +197,21 @@ def compute_up(t, x, **kwargs):
             keepdims=t.keepdims)
 
 
+@compute_up.register(Distinct, np.recarray)
+def recarray_distinct(t, rec, **kwargs):
+    return pd.DataFrame.from_records(rec).drop_duplicates(
+        subset=t.on or None).to_records(index=False).astype(rec.dtype)
+
+
 @dispatch(Distinct, np.ndarray)
-def compute_up(t, x, **kwargs):
-    return np.unique(x)
+def compute_up(t, arr, _recarray_distinct=recarray_distinct, **kwargs):
```

Comment:
Yeah, it swaps out a global lookup for a local lookup. It's not a huge gain but it does slightly cut down the cost of calling the function.